core/src/test/scala/unit/kafka/log/LogCleanerIntegrationTest.scala

```diff
@@ -228,6 +231,60 @@ class LogCleanerIntegrationTest(compressionCodec: String) extends AbstractLogCle
     checkLogAfterAppendingDups(log, startSize, appends)
   }
 
+  @Test(timeout = 60000)
+  def testMarksPartitionsAsOfflineAndPopulatesUncleanableMetrics() {
+    val largeMessageKey = 20
+    val (_, largeMessageSet) = createLargeSingleMessageSet(largeMessageKey, RecordBatch.CURRENT_MAGIC_VALUE)
+    val maxMessageSize = largeMessageSet.sizeInBytes
+    cleaner = makeCleaner(partitions = topicPartitions, maxMessageSize = maxMessageSize)
+
+    def breakPartitionLog(tp: TopicPartition): Unit = {
+      val log = cleaner.logs.get(tp)
+      writeDups(numKeys = 100, numDups = 3, log = log, codec = codec)
+
+      val partitionFile = log.logSegments.last.log.file()
+      val writer = new PrintWriter(partitionFile)
+      writer.write("jogeajgoea")
+      writer.close()
+
+      writeDups(numKeys = 100, numDups = 3, log = log, codec = codec)
+    }
+
+    def getGauge[T](metricName: String, metricScope: String): Gauge[T] = {
+      Metrics.defaultRegistry.allMetrics.asScala
+        .filterKeys(k => {
+          k.getName.endsWith(metricName) && k.getScope.endsWith(metricScope)
+        })
+        .headOption
+        .getOrElse { fail(s"Unable to find metric $metricName") }
+        .asInstanceOf[(Any, Gauge[Any])]
+        ._2
+        .asInstanceOf[Gauge[T]]
+    }
+
+    breakPartitionLog(topicPartitions(0))
+    breakPartitionLog(topicPartitions(1))
+
+    cleaner.startup()
+
+    val log = cleaner.logs.get(topicPartitions(0))
+    val log2 = cleaner.logs.get(topicPartitions(1))
+    val uncleanableDirectory = log.dir.getParent
+    val uncleanablePartitionsCountGauge = getGauge[Int]("uncleanable-partitions-count", uncleanableDirectory)
+    val uncleanableBytesGauge = getGauge[Long]("uncleanable-bytes", uncleanableDirectory)
+
+    TestUtils.waitUntilTrue(() => uncleanablePartitionsCountGauge.value() == 2, "There should be 2 uncleanable partitions", 55000L)
```

Comment:
Does the test need to wait for 55 secs? It seems that 10 secs will be a lot.

Reply:
Do you know why it's taking 20 secs? Could we reduce the number of keys to make the test faster?

Reply:
I found out - it's the backoff config for the cleaner. Lowering it and the number of keys written causes the test to finish in ~1.5s. Awesome!